src/overlays/actors/ovl_En_Po_Relay/z_en_po_relay.c

```diff
@@ -21,35 +44,367 @@ const ActorInit En_Po_Relay_InitVars = {
+    this->hookshotSlotFull = INV_CONTENT(ITEM_HOOKSHOT) != ITEM_NONE;
+    this->unk_19A = func_8002DAC0(&this->actor, &vec);
+    this->actor.flags |= 0x8000000;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_LAUGH);
+    this->actionFunc = EnPoRelay_Race;
+}
+
+void EnPoRelay_SetupEndRace(EnPoRelay* this) {
+    this->actor.posRot.rot.y = this->actor.initPosRot.rot.y + 0xC000;
+    this->actor.flags &= ~0x08000000;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = EnPoRelay_EndRace;
+}
+
+void EnPoRelay_CorrectY(EnPoRelay* this) {
+    Math_ApproxF(&this->actor.initPosRot.pos.y, D_80AD8C30[(this->pathIndex >= 28) ? 27 : this->pathIndex].y + 45.0f,
+                 2.0f);
+    this->actor.posRot.pos.y = Math_Sins(this->unk_195 * 0x800) * 8.0f + this->actor.initPosRot.pos.y;
+}
+
+void EnPoRelay_Idle(EnPoRelay* this, GlobalContext* globalCtx) {
+    Math_ApproxUpdateScaledS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 0x100);
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        // Triggers the frame after link is in range
+        this->actor.flags &= ~0x10000;
+        this->actionFunc = EnPoRelay_Talk;
+    } else if (this->actor.xzDistFromLink < 250.0f) {
+        // Wait for link to get in range
+        this->actor.flags |= 0x10000;
+        this->actor.textId = this->textId;
+        func_8002F2CC(&this->actor, globalCtx, 250.0f);
+    }
+    func_8002F974(&this->actor, NA_SE_EN_PO_FLY - SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Po_Relay/func_80AD790C.s")
+void EnPoRelay_Talk(EnPoRelay* this, GlobalContext* globalCtx) {
+    Math_ApproxUpdateScaledS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 0x100);
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        Actor_SetTextWithPrefix(globalCtx, &this->actor, 0x2F);
+        this->textId = this->actor.textId;
+        EnPoRelay_SetupRace(this);
+    }
+    func_8002F974(&this->actor, NA_SE_EN_PO_FLY - SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Po_Relay/func_80AD7944.s")
+#ifdef NON_MATCHING
+// Single stack difference
+void EnPoRelay_Race(EnPoRelay* this, GlobalContext* globalCtx) { // saved, sp64
+    Player* player = PLAYER;                                     // sp5C
+    Vec3f vec;                                                   // sp50
+    f32 rand;
+    f32 multiplier; // sp48
+    f32 speed;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Po_Relay/func_80AD7984.s")
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (this->actionTimer == 0 && Math_Rand_ZeroOne() < 0.03f) {
+        this->actionTimer = 32;
+        if (this->pathIndex < 23) {
+            rand = Math_Rand_ZeroOne() * 3.0f;
+            if (rand < 1.0f) {
+                multiplier = 1.0f;
+            } else if (rand < 2.0f) {
+                multiplier = -1.0f;
+            } else {
+                multiplier = 0.0f;
+            }
```

Comment:
i know this is non matching but id call `multiplier` as direction, and maybe note which one is center/left/right. relative to link or dampe? i dunno. relative to link would be easier to understand i guess since thats the perspective of the player when looking at the flame. i leave that up to you

Reply:
oh nvm i just saw that multiplier is reused later so this cant work. a shame. some comments about the direction will help i suppose